File: Python/8py 3/reverse_list.py
"""
|--------------------------------------------------------------------------
| Reverse List Order
|--------------------------------------------------------------------------
|
"""

def reverse_list(l):
   l.reverse()
   return l


def reverse_list(l):
   return [r for r in reversed(l)]
   

def reverse_list(l):
   return l[::-1]


def reverse_list(l):
   return list(reversed(l))


print(reverse_list([1,2,3,4]), [4,3,2,1])
print(reverse_list([3,1,5,4]), [4,5,1,3])
print(reverse_list([3,6,9,2]), [2,9,6,3])
print(reverse_list([1]), [1])

File: PHP/8php 2/reverseList.php
<?php
/*
|--------------------------------------------------------------------------
| Reverse List Order
|--------------------------------------------------------------------------
|
*/

function reverseList(array $list): array {
   return array_reverse($list);
}


function reverseList(array $list): array {
   $result = [];
   for($i = count($list) - 1; $i >= 0; $i--) {
      $result[] = $list[$i];
   }
   return $result;
}


function reverseList(array $list): array {
   $i = 0; $arrayLen = count($list); $x = $arrayLen - 1; $newList = array();
   while($i < $arrayLen) {
      array_push($newList,$list[$x]);
      $x--;
      $i++;
   }
   return ($newList);
}


print_r(reverseList([1, 2, 3, 4]));
print_r(reverseList([3, 1, 5, 4]));